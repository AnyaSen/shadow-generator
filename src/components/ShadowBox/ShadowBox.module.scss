@import "../../assets/scss/variables";

.ShadowBoxContainer {
  display: flex;
  justify-content: center;
  align-items: center;
  width: 60%;

  .ShadowBox {
    height: 45rem;
    width: 45rem;
    margin: $space-1;
    box-shadow: $shadow-1;
    border-radius: 1rem;
    background-color: #f3f3f3;
  }
}
