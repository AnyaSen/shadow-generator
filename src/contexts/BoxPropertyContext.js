import React, { useState, createContext } from "react";

export const BoxPropertyContext = createContext();

export const BoxPropertyProvider = ({ children }) => {
  const [heightValue, setHeightValue] = useState(350);
  const [widthValue, setWidthValue] = useState(350);
  const [radiusValue, setRadiusValue] = useState(16);
  const [borderValue, setBorderValue] = useState(0);

  const ChangeHeightValue = (event, newValue) => {
    setHeightValue(newValue);
  };

  const ChangeWidthValue = (event, newValue) => {
    setWidthValue(newValue);
  };

  const ChangeBorderValue = (event, newValue) => {
    setBorderValue(newValue);
  };

  const ChangeRadiusValue = (event, newValue) => {
    setRadiusValue(newValue);
  };

  return (
    <BoxPropertyContext.Provider
      value={{
        heightValue,
        ChangeHeightValue,
        widthValue,
        ChangeWidthValue,
        borderValue,
        ChangeBorderValue,
        radiusValue,
        ChangeRadiusValue
      }}
    >
      {children}
    </BoxPropertyContext.Provider>
  );
};
