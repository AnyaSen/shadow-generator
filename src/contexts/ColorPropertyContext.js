import React, { useState, createContext } from "react";

export const ColorPropertyContext = createContext();

export const ColorPropertyProvider = ({ children }) => {
  const [boxColor, setBoxColor] = useState("#f3f3f3");
  const [shadowColor, setShadowColor] = useState("#999999");
  const [bgColor, setBgColor] = useState("#fff");
  const [borderColor, setBorderColor] = useState("#808080");

  const ChangeBoxColor = newColor => {
    setBoxColor(newColor.hex);
  };
  const ChangeShadowColor = newColor => {
    setShadowColor(newColor.hex);
  };
  const ChangeBgColor = newColor => {
    setBgColor(newColor.hex);
  };
  const ChangeBorderColor = newColor => {
    setBorderColor(newColor.hex);
  };

  return (
    <ColorPropertyContext.Provider
      value={{
        boxColor,
        ChangeBoxColor,
        ChangeShadowColor,
        shadowColor,
        ChangeBgColor,
        bgColor,
        ChangeBorderColor,
        borderColor
      }}
    >
      {children}
    </ColorPropertyContext.Provider>
  );
};
