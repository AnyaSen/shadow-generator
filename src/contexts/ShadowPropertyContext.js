import React, { useState, createContext } from "react";

export const ShadowPropertyContext = createContext();

export const ShadowPropertyProvider = ({ children }) => {
  const [shiftHorizontically, setShiftHorizontically] = useState(20);
  const [shiftVertically, setShiftVertically] = useState(14);
  const [blurValue, setBlurValue] = useState(10);
  const [opacityValue, setOpacityValue] = useState(12);

  const ChangeShiftHorizontically = (event, newValue) => {
    setShiftHorizontically(newValue);
  };

  const ChangeShiftVertically = (event, newValue) => {
    setShiftVertically(newValue);
  };

  const ChangeBlurValue = (event, newValue) => {
    setBlurValue(newValue);
  };

  const ChangeOpacityValue = (event, newValue) => {
    setOpacityValue(newValue);
  };

  return (
    <ShadowPropertyContext.Provider
      value={{
        shiftHorizontically,
        ChangeShiftHorizontically,
        shiftVertically,
        ChangeShiftVertically,
        blurValue,
        ChangeBlurValue,
        opacityValue,
        ChangeOpacityValue
      }}
    >
      {children}
    </ShadowPropertyContext.Provider>
  );
};
