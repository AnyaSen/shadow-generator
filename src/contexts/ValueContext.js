import React, { useState, createContext } from "react";

export const ValueContext = createContext();

export const ValueProvider = ({ children }) => {
  const [heightValue, setHeightValue] = useState(45);
  const [widthValue, setWidthValue] = useState(45);
  const [radiusValue, setRadiusValue] = useState(1);
  const [shiftHorizontically, setShiftHorizontically] = useState(1.25);
  const [shiftVertically, setShiftVertically] = useState(0.85);
  const [blurValue, setBlurValue] = useState(0.625);
  const [opacityValue, setOpacityValue] = useState(0.75);
  const [borderValue, setBorderValue] = useState(0);
  const [boxColor, setBoxColor] = useState("#f3f3f3");
  const [shadowColor, setShadowColor] = useState("#999999");
  const [bgColor, setBgColor] = useState("#fff");
  const [borderColor, setBorderColor] = useState("#808080");

  const ChangeHeightValue = (event, newValue) => {
    setHeightValue(newValue);
  };
  const ChangeWidthValue = (event, newValue) => {
    setWidthValue(newValue);
  };
  const ChangeBorderValue = (event, newValue) => {
    setBorderValue(newValue);
  };
  const ChangeRadiusValue = (event, newValue) => {
    setRadiusValue(newValue);
  };
  const ChangeShiftHorizontically = (event, newValue) => {
    setShiftHorizontically(newValue);
  };
  const ChangeShiftVertically = (event, newValue) => {
    setShiftVertically(newValue);
  };
  const ChangeBlurValue = (event, newValue) => {
    setBlurValue(newValue);
  };
  const ChangeOpacityValue = (event, newValue) => {
    setOpacityValue(newValue);
  };
  const ChangeBoxColor = newColor => {
    setBoxColor(newColor.hex);
  };
  const ChangeShadowColor = newColor => {
    setShadowColor(newColor.hex);
  };
  const ChangeBgColor = newColor => {
    setBgColor(newColor.hex);
  };
  const ChangeBorderColor = newColor => {
    setBorderColor(newColor.hex);
  };

  return (
    <ValueContext.Provider
      value={{
        heightValue,
        ChangeHeightValue,
        widthValue,
        ChangeWidthValue,
        borderValue,
        ChangeBorderValue,
        radiusValue,
        ChangeRadiusValue,
        shiftHorizontically,
        ChangeShiftHorizontically,
        shiftVertically,
        ChangeShiftVertically,
        blurValue,
        ChangeBlurValue,
        opacityValue,
        ChangeOpacityValue,
        boxColor,
        ChangeBoxColor,
        ChangeShadowColor,
        shadowColor,
        ChangeBgColor,
        bgColor,
        ChangeBorderColor,
        borderColor
      }}
    >
      {children}
    </ValueContext.Provider>
  );
};
