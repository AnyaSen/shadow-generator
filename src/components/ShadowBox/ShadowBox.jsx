import React, { useContext } from "react";
import Styles from "./ShadowBox.module.scss";
import hexToRgba from "hex-to-rgba";

import { BoxPropertyContext } from "../../contexts/BoxPropertyContext";
import { ColorPropertyContext } from "../../contexts/ColorPropertyContext";
import { ShadowPropertyContext } from "../../contexts/ShadowPropertyContext";

export default function ShadowBox() {
  const { heightValue, widthValue, borderValue, radiusValue } = useContext(
    BoxPropertyContext
  );
  const {
    shiftHorizontically,
    shiftVertically,
    opacityValue,
    blurValue
  } = useContext(ShadowPropertyContext);

  const { boxColor, shadowColor, borderColor, bgColor } = useContext(
    ColorPropertyContext
  );

  return (
    <div
      className={Styles.ShadowBoxContainer}
      style={{ backgroundColor: bgColor }}
    >
      <div
        style={{
          height: `${heightValue}px`,
          width: `${widthValue}px`,
          boxShadow: `${shiftHorizontically}px ${shiftVertically}px ${blurValue}px ${hexToRgba(
            shadowColor,
            opacityValue
          )}`,
          border: `${borderValue}px solid ${borderColor}`,
          borderRadius: `${radiusValue}px`,
          backgroundColor: boxColor
        }}
        className={Styles.ShadowBox}
      ></div>
    </div>
  );
}
