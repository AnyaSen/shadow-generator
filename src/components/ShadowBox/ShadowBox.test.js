import React from "react";
import { render, cleanup } from "@testing-library/react";

import ShadowBox from "./ShadowBox";

import { BoxPropertyProvider } from "../../contexts/BoxPropertyContext";
import { ShadowPropertyProvider } from "../../contexts/ShadowPropertyContext";
import { ColorPropertyProvider } from "../../contexts/ColorPropertyContext";

describe("<ShadowBox />", () => {
  afterEach(cleanup);

  test("Should have have a child.", () => {
    const { getByTestId, container } = render(
      <BoxPropertyProvider>
        <ShadowPropertyProvider>
          <ColorPropertyProvider>
            <ShadowBox />
          </ColorPropertyProvider>
        </ShadowPropertyProvider>
      </BoxPropertyProvider>
    );

    const ShadowBoxContainer = getByTestId("Box");

    expect(ShadowBoxContainer.children.length).toEqual(1);

    expect(container.firstChild).toMatchSnapshot();
  });
});
