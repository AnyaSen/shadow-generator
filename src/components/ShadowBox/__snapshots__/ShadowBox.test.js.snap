// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ShadowBox /> Should have have a child. 1`] = `
<div
  class="ShadowBoxContainer"
  data-testid="Box"
  style="background-color: rgb(255, 255, 255);"
>
  <div
    class="ShadowBox"
    style="height: 200px; width: 200px; box-shadow: 20px 14px 10px rgba(153, 153, 153, 12); border: 0px solid #808080; border-radius: 16px; background-color: rgb(243, 243, 243);"
  />
</div>
`;
